processing/src/main/java/io/druid/query/BaseQuery.java

```diff
@@ -34,10 +35,66 @@
 
 /**
  */
-public abstract class BaseQuery<T> implements Query<T>
+public abstract class BaseQuery<T extends Comparable<T>> implements Query<T>
 {
+  public static <T> int getContextPriority(Query<T> query, int defaultValue)
```

Comment:
all those method are marked as `@deprecated` in the Query interface, should we mark them deprecated here too?